ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/SandboxLedgerBackend.scala

```diff
@@ -260,4 +271,48 @@ class SandboxLedgerBackend(ledger: Ledger)(implicit mat: Materializer)
     getTransactionById(transactionId.unwrap)
       .map(_.flatMap(TransactionConversion.acceptedToDomainTree(_, filter)))(DEC)
   }
+
+  override def getCompletions(
+      begin: LedgerOffset,
+      applicationId: ApplicationId,
+      parties: Set[Ref.Party]
+  ): Source[CompletionEvent, NotUsed] = {
+    val converter = new OffsetConverter()
+    converter.toAbsolute(begin).flatMapConcat {
+      case LedgerOffset.Absolute(absBegin) =>
+        ledgerSyncEvents(Some(absBegin)).collect {
+          case at: LedgerSyncEvent.AcceptedTransaction =>
+            CommandAccepted(
+              domain.LedgerOffset.Absolute(at.offset),
+              at.recordTime,
+              at.commandId.map(domain.CommandId(_)),
```

Comment:
If the `AcceptedTransaction` doesn't have the commandId set, it means we received the transaction from some other node.
It seems that the filtering on applicationId and party is missing, or did I overlook it?

Reply:
you did not, it's indeed missing. fix is on the way

Reply:
it's pushed